packages/devextreme/js/__internal/scheduler/appointments/m_appointment_collection.ts

```diff
@@ -832,24 +839,32 @@ class SchedulerAppointments extends CollectionWidget {
   _correctStartDateByDelta(startDate, deltaTime) {
     const endDayHour = this.invoke('getEndDayHour');
     const startDayHour = this.invoke('getStartDayHour');
-    let result = startDate.getTime() - deltaTime;
-    const visibleDayDuration = (endDayHour - startDayHour) * toMs('hour');
-
-    const daysCount = deltaTime > 0
-      ? Math.ceil(deltaTime / visibleDayDuration)
-      : Math.floor(deltaTime / visibleDayDuration);
 
     const maxDate = new Date(startDate);
     const minDate = new Date(startDate);
 
     minDate.setHours(startDayHour, 0, 0, 0);
     maxDate.setHours(endDayHour, 0, 0, 0);
 
+    const correctStartDate = new Date(startDate);
+
+    if (correctStartDate.getTime() < minDate.getTime()) {
+      correctStartDate.setHours(startDayHour, 0, 0, 0);
+    }
```

Comment:
Minor comment.

Let's reorganize this code a little:
```suggestion
    const maxDate = new Date(startDate);
    const minDate = new Date(startDate);
    const correctStartDate = new Date(startDate);

    minDate.setHours(startDayHour, 0, 0, 0);
    maxDate.setHours(endDayHour, 0, 0, 0);

    if (correctStartDate < minDate) {
      correctStartDate.setHours(startDayHour, 0, 0, 0);
    }
```